build/templates/job.template.yaml

```diff
@@ -61,16 +94,8 @@ spec:
           value: /must-gather
         - name: must_gather_upload
           value: /must-gather-upload
-      initContainers:
-{{- range $index, $element := .Spec.MustGatherImages }}
-      - command:
-        - /usr/bin/gather
-        image: {{ $element }}
-        name: gather-{{ $index }}
-        volumeMounts:
-        - mountPath: /must-gather
-          name: must-gather-output
-{{- end }}
```

Comment:
@dustman9000 most of the changes in this file involve making the container that runs the gather script a normal container instead of an initContainer. this allows us to collect logs from the gather script and see what's going wrong, if anything. Benson added some logic to the upload container that allows it to wait until the gather is completed before it attempts to upload the must-gather, since the containers now run concurrently.

there's also logic that adds a timeout field to the CR (as I originally erroneously thought the error in OSD-8230 was due to a timeout issue). this defaults to 0 (no timeout) and is optional